### DISCUSSION TITLE
Possible regulation threshold issue

### DISCUSSION THREAD

**Discussion by Edwintenhaaf** [jmcollin78/versatile_thermostat#375] [Q&A] [answered]
Hi All,

Before creating an issue I want to post it here to check  if the cause is related to versatile_thermostat.
Im using thermostat_over_climate to control three toshiba ac's.
I have configured the step size to 1 degree but I get these errors in the log.
The thermostat does not accept the temperatures because it only accepts rounded degrees.

the climate entities have the correct settings/attributes

min_temp: 15
max_temp: 30
target_temp_step: 1

Does any of you know what the problem can be ?

`Logger: homeassistant.core
Source: core.py:2295
First occurred: 12:55:56 PM (16 occurrences)
Last logged: 6:32:12 PM

Error executing service: <ServiceCall climate.set_temperature (c:01HP4SJ3PG5VS7ZXG7B6EGDC0C): entity_id=['climate.woonkamer_ac'], temperature=**25.2**, target_temp_high=30.0, target_temp_low=15.0>
Error executing service: <ServiceCall climate.set_temperature (c:01HP4SJ3PZ3G2VNK43WYFDSHNH): entity_id=['climate.overloop_ac'], temperature=**23.6**, target_temp_high=30.0, target_temp_low=15.0>
Error executing service: <ServiceCall climate.set_temperature (c:01HP4SW6Z0MKQ5FDDPJ7BT9Z38): entity_id=['climate.woonkamer_ac'], temperature=**24.27**, target_temp_high=30.0, target_temp_low=15.0>
Error executing service: <ServiceCall climate.set_temperature (c:01HP4V9BK46E101DWECWK7JPYN): entity_id=['climate.woonkamer_ac'], temperature=**23.01**, target_temp_high=30.0, target_temp_low=15.0>
Error executing service: <ServiceCall climate.set_temperature (c:01HP4WDW6J7C3SPG45PF77F0GB): entity_id=['climate.overloop_ac'], temperature=**23.8**, target_temp_high=30.0, target_temp_low=15.0>
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/core.py", line 2295, in _run_service_call_catch_exceptions
    await coro_or_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 2316, in _execute_service
    return await target(service_call)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 892, in entity_service_call
    single_response = await _handle_entity_call(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 962, in _handle_entity_call
    result = await task
             ^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/climate/__init__.py", line 835, in async_service_temperature_set
    await entity.async_set_temperature(**kwargs)
  File "/config/custom_components/toshiba_ac/climate.py", line 112, in async_set_temperature
    await self._device.set_ac_temperature(set_temperature)
  File "/usr/local/lib/python3.12/site-packages/toshiba_ac/device/__init__.py", line 289, in set_ac_temperature
    state.ac_temperature = val
    ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/toshiba_ac/device/fcu_state.py", line 393, in ac_temperature
    self._ac_temperature = ToshibaAcFcuState.AcTemperature.to_raw(val)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/toshiba_ac/device/fcu_state.py", line 50, in to_raw
    return temp_to_raw[temperature]
           ~~~~~~~~~~~^^^^^^^^^^^^^
KeyError: 23.05

`

Thanks,

Edwin

**Comment by jmcollin78** [comment]
Hello @Edwintenhaaf ,

If the underlying climate doesn't support .1 degree step you can set the self_regulation_threshold to 1.

![Capture d’écran 2024-02-09 à 00 22 21](https://github.com/jmcollin78/versatile_thermostat/assets/1712374/49cd424a-7a4c-449b-aefe-79bb5c831d8f)

The step is used to increment or decrement the target temperature but is not used to round the value sends to the underlying.

**Reply by Edwintenhaaf** [reply]
Hi @jmcollin78, 

Thanks for you're versatile_thermostat, it is a must have when heating with aicrconditioners.
I must change the title because we are talking about the self_regulation_threshold.
<img width="494" alt="image" src="https://github.com/jmcollin78/versatile_thermostat/assets/51457250/e4cdc108-ff26-4057-b9f6-026f063e97eb">
As you can see this is allready set to 1

fixed it for now with a autmation that syncs de regulated temperature with the ac.

`alias: "temp fix: regulated temp set on ac"
if:
  - condition: template
    value_template: >-
      {% set regulated = states('sensor.woonkamer_regulated_temperature') |
      round %}

      {% set temperature =state_attr('climate.woonkamer_ac','temperature') %}

      {% if regulated != temperature %}true{%endif%}
then:
  - service: climate.set_temperature
    metadata: {}
    data:
      hvac_mode: heat
      temperature: "{{ states('sensor.woonkamer_regulated_temperature') | round }}"
    target:
      entity_id: climate.woonkamer_ac
enabled: true`

**Reply by Edwintenhaaf** [reply]
I just downgraded to 5.3.3 and then the issue disapeared.
Can it be that this change is causing the issue ?
`Issue #348 - self regulation use internal temp by @jmcollin78 in https://github.com/jmcollin78/versatile_thermostat/pull/373. Note that this new feature should reduce the need of Strong Regulation`

**Comment by maia** [comment]
Unsure if this is related, but since upgrading HA to 2024.2.0 my climate device lovelace UI will increase/decrease by 0.1°C instead of 0.5°C as it should be. As I found out, there are two places one can configure the step size: In configuration there is "regulation threshold" and in "central configuration" there is "temperature step". Interestingly the "temperature step" was set to 0.1°C and I have no explanation why, because until recently a manual change of the target temperature always went in 0.5°C steps. So maybe it's also caused by a new version of VTherm and not the HA release?

@jmcollin78 As a hint, for people who do not speak french it's difficult if you show a french screenshot and then the internal variable name (self_regulation_threshold), as that isn't displayed anywhere in the frontend, it's only known to you.

**Comment by jmcollin78** [comment]
Humm, I will have a look. It is possible that the new feature (#348) have an impact on the `self_regulation_threshold` behavior

**Comment by jmcollin78** [comment, marked as answer]
It has been fixed in this release: https://github.com/jmcollin78/versatile_thermostat/releases/tag/5.4.2

**Reply by Edwintenhaaf** [reply]
Good to know that you first have to disable the 'use internal termperature of the underlying' setting and finish the configuration. Then configure it again to enable it. At least in my case that did the trick to get it working.
![image](https://github.com/jmcollin78/versatile_thermostat/assets/51457250/32bd28cf-c844-4f99-b069-a84496e0a334)
